File: semestre3/anderson/organizar dps/pilha.py
class Pilha:
    def __init__(self):
        self.itens = []

    def push(self, item):
        self.itens.append(item)
    
    def pop(self):
        if not self.is_empty():
            return self.itens.pop()
        else:
            return None
    
    def is_empty(self):
        return len(self.itens) == 0
    
    def __str__(self):
        return f"Pilha: {self.itens}"
    
    def tamanho(self):
        return len(self.itens)
    
    def topo(self):
        if not self.is_empty:
            return self.itens[-1]
        else:
            return None
    
pilha = Pilha()

pilha.push('copo')

pilha.is_empty()

File: semestre3/anderson/pilha.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 100

typedef struct
{
    int itens[MAX];
    int topo;
} Pilha;

void inicializarPilha(Pilha *p)
{
    p->topo = -1;
}

int estaVazia(Pilha *p)
{
    return (p->topo = -1);
}

int estaCheia(Pilha *p)
{
    return (p->topo == MAX - 1);
}

void push(Pilha *p)
{
    if*(estaCheia(p)){
        printf("Erro: Pilha cheia");
        return;
    }
    
    p->topo++;
    p->itens[p->topo] = valor;
}

int pop(Pilha *p)
{
    if(estaVazia(p)){
        printf("Erro: Pilha vazia");
        return;
    }

    int valor = p->itens[p->topo];
    p->topo--;
    return valor;
}

int peek(Pilha *p)
{
    if(estaVazia(p)){
        printf("Erro: Pilha vazia");
        return;
    }

    return p->itens[p->topo];
}

int main(void)
{
    int num = 34;
    int resultado = fatorial(num);
    printf("O fatorial de %d é %d", num, resultado);
    return 0;
}
